src/GiftedChat.tsx

```diff
@@ -232,360 +232,233 @@ export interface GiftedChatState<TMessage extends IMessage = IMessage> {
-    renderActions: PropTypes.func,
-    renderSend: PropTypes.func,
-    renderAccessory: PropTypes.func,
-    onPressActionButton: PropTypes.func,
-    bottomOffset: PropTypes.number,
-    minInputToolbarHeight: PropTypes.number,
-    listViewProps: PropTypes.object,
-    keyboardShouldPersistTaps: PropTypes.oneOf(['always', 'never', 'handled']),
-    onInputTextChanged: PropTypes.func,
-    maxInputLength: PropTypes.number,
-    forceGetKeyboardHeight: PropTypes.bool,
-    inverted: PropTypes.bool,
-    textInputProps: PropTypes.object,
-    extraData: PropTypes.object,
-    minComposerHeight: PropTypes.number,
-    maxComposerHeight: PropTypes.number,
-    alignTop: PropTypes.bool,
-  }
-
-  static append<TMessage extends IMessage>(
-    currentMessages: TMessage[] = [],
-    messages: TMessage[],
+    onInputTextChanged = null,
+    maxInputLength = null,
+    forceGetKeyboardHeight = false,
     inverted = true,
-  ) {
-    if (!Array.isArray(messages)) {
-      messages = [messages]
-    }
-    return inverted
-      ? messages.concat(currentMessages)
-      : currentMessages.concat(messages)
-  }
-
-  static prepend<TMessage extends IMessage>(
-    currentMessages: TMessage[] = [],
-    messages: TMessage[],
-    inverted = true,
-  ) {
-    if (!Array.isArray(messages)) {
-      messages = [messages]
-    }
-    return inverted
-      ? currentMessages.concat(messages)
-      : messages.concat(currentMessages)
-  }
-
-  _isMounted: boolean = false
-  _keyboardHeight: number = 0
-  _bottomOffset: number = 0
-  _maxHeight?: number = undefined
-  _isFirstLayout: boolean = true
-  _locale: string = 'en'
-  invertibleScrollViewProps: any = undefined
-  _actionSheetRef: RefObject<ActionSheetProviderRef> = React.createRef()
-  _messageContainerRef?: RefObject<FlatList<IMessage>> = React.createRef()
-  _isTextInputWasFocused: boolean = false
-  textInput?: any
-
-  state = {
+    extraData = null,
+    minComposerHeight = MIN_COMPOSER_HEIGHT,
+    maxComposerHeight = MAX_COMPOSER_HEIGHT,
+  } = props
+
+  const _isMounted = useRef(false)
+  const _keyboardHeight = useRef(0)
+  const _bottomOffset = useRef(0)
+  const _maxHeight = useRef<number | undefined>(undefined)
+  const _isFirstLayout = useRef(true)
+  let _locale: string = 'en'
+  let _actionSheetRef = useRef<ActionSheetProviderRef>()
+  let _messageContainerRef = useRef<FlatList<IMessage> | null>()
+  let _isTextInputWasFocused: boolean = false
+  let textInputRef = useRef<TextInput>()
+
+  const [state, setState] = useState<GiftedChatState>({
     isInitialized: false, // initialization will calculate maxHeight before rendering the chat
-    composerHeight: this.props.minComposerHeight,
+    composerHeight: minComposerHeight,
     messagesContainerHeight: undefined,
     typingDisabled: false,
     text: undefined,
     messages: undefined,
-  }
+  })
 
-  constructor(props: GiftedChatProps<TMessage>) {
-    super(props)
-
-    this.invertibleScrollViewProps = {
```

Comment:
Moved this to directly where it's being used. https://github.com/FaridSafi/react-native-gifted-chat/pull/2372/files#diff-b11e3f8b30e19689a33f545b3567228eb41fd9cfbbad474b1b21e7b63eaa2d57R567